data/data/gcp/network/common.tf

```diff
@@ -1,2 +1,6 @@
 # Canonical internal state definitions for this module.
 # read only: only locals and data source definitions allowed. No resources or module blocks in this file
+
+locals {
+  cluster_network = var.preexisting_cluster_network == "" ? google_compute_network.cluster_network[0].self_link : var.preexisting_cluster_network
```

Comment:
I wish TF allowed ternary to be split onto separate lines

Reply:
Perhaps you can set a local variable once and use it throughout?